code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/ProjectMemberResourceManager.java

```diff
@@ -64,20 +73,19 @@ protected Object listInternal(SchemaFactory schemaFactory, String type, Map<Obje
             return Arrays.asList(projectMember);
         }
         String projectId = (String) criteria.get("projectId");
+        List<? extends ProjectMember> members;
         if (StringUtils.isNotEmpty(projectId)) {
-            List<? extends ProjectMember> members = authDao.getActiveProjectMembers(Long.valueOf(projectId));
-            for (ProjectMember member: members){
-                member = untransform(member);
-                policy.grantObjectAccess(member);
-            }
-            return members;
+            members =  authDao.getActiveProjectMembers(Long.valueOf(projectId));
+        } else {
+            members = authDao.getActiveProjectMembers(policy.getAccountId());
         }
-        List<? extends ProjectMember> members = authDao.getActiveProjectMembers(policy.getAccountId());
+        List<ProjectMember> membersToReturn = new ArrayList<>();
         for (ProjectMember member: members){
             member = untransform(member);
+            membersToReturn.add(member);
```

Comment:
This allows easy masking of member types. I could check for null and not add nulls to the list though.